File: app/src/main/java/com/lakue/newsreader/Base/BaseUtil.java
package com.lakue.newsreader.Base;

import android.content.Context;
import android.util.DisplayMetrics;
import android.util.TypedValue;

public class BaseUtil {

    /**
     *
     * @param px
     * @param context
     * @return
     */
    public static int convertPixelsToDp(float px, Context context) {
        return (int) TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, px, context.getResources().getDisplayMetrics());
    }

    /**
     *
     * @param context
     * @param dp
     * @return
     */
    public static float dpToPx(Context context, float dp) {
        DisplayMetrics dm = context.getResources().getDisplayMetrics();
        return (int) TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, dp, dm);
    }

    public static int getDeviceWidth(Context context){
        DisplayMetrics dm = context.getResources().getDisplayMetrics();
        return dm.widthPixels;
    }

    public static int getDeviceHeight(Context context){
        DisplayMetrics dm = context.getResources().getDisplayMetrics();
        return dm.heightPixels;
    }

}

File: app/src/main/java/com/lakue/newsreader/Base/BaseUtil.kt
package com.lakue.newsreader.Base

import android.content.Context
import android.util.DisplayMetrics
import android.util.TypedValue

object BaseUtil {

    /**
     * px to dp
     * @param px
     * @param context
     * @return
     */
    fun convertPixelsToDp(px: Float, context: Context): Int {
        return TypedValue.applyDimension(
            TypedValue.COMPLEX_UNIT_DIP,
            px,
            context.resources.displayMetrics
        ).toInt()
    }

    /**
     * dp to px
     * @param context
     * @param dp
     * @return
     */
    fun dpToPx(context: Context, dp: Float): Float {
        val dm = context.resources.displayMetrics
        return TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, dp, dm).toInt().toFloat()
    }

    /**
     * get device width
     * @param context
     * @return
     */
    fun getDeviceWidth(context: Context): Int {
        val dm = context.resources.displayMetrics
        return dm.widthPixels
    }

    /**
     * get device height
     * @param context
     * @return
     */
    fun getDeviceHeight(context: Context): Int {
        val dm = context.resources.displayMetrics
        return dm.heightPixels
    }

}
